components/Home/VeOlas/ManageYourLock/IncreaseAmount.jsx

```diff
@@ -0,0 +1,104 @@
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Button, Form, Typography, Divider,
+} from 'antd/lib';
+import {
+  notifyError,
+  notifySuccess,
+  CannotIncreaseAlert,
+  AlreadyAllAmountLocked,
+} from 'common-util/functions';
+import {
+  fetchMappedBalances,
+  fetchVotesAndTotalSupplyLocked,
+} from 'store/setup/actions';
+import { parseAmount, FormItemInputNumber } from '../../common';
+import { updateIncreaseAmount } from '../utils';
+
+const { Title } = Typography;
+
+export const IncreaseAmount = () => {
+  const dispatch = useDispatch();
+  const account = useSelector((state) => state?.setup?.account);
+  const chainId = useSelector((state) => state?.setup?.chainId);
+  const isMappedAmountZero = useSelector(
+    (state) => state?.setup?.mappedBalances?.isMappedAmountZero || false,
+  );
+  const olasBalance = useSelector((state) => state?.setup?.olasBalance);
+  const hasNoOlasBalance = Number(olasBalance || '0') === 0;
+
+  /**
+   * can increase amount only if the mapped amount is zero (ie. no lock exists)
+   * or if the user has some olas tokens.
+   */
+  const cannotIncreaseAmount = isMappedAmountZero || hasNoOlasBalance || !account;
```

Comment:
Not sure I understand this. THe account can increase if there's already a lock in veOLAS, cannot lock if there is no veOLAS locked. Also, is there a check against the input OLAS amount that the account wants to increase the lock for? There's a check on-chain, wondering if there's one off-chain as well.

Reply:
There could be a check for `OLAS.balanceOf(account)` - if it is less than or equal than the input value for the increase amount. This way the user would not be able to input more OLAS to increase the lock for than they actually have. The same could be applied (or it is already applied) to the `createLock()` function.